spec/lib/adv_ctc_irs1040_pdf_spec.rb

```diff
@@ -0,0 +1,255 @@
+                                    "TaxableSocSecAmt6b" => nil,
+                                    "TotalAdditionalIncomeAmt8" => nil,
+                                    "TotalAdjustmentsAmt10a" => nil,
+                                    "TotalAdjustmentsToIncomeAmt10c" => nil,
+                                    "TotalCreditsAmt21" => nil,
+                                    "TotalDeductionsAmt14" => nil,
+                                    "TotalIncomeAmt9" => "1",
+                                    "TotalItemizedOrStandardDedAmt12" => "",
+                                    "TotalNonrefundableCreditsAmt20" => nil,
+                                    "TotalOtherPaymentsRfdblCrAmt31" => nil,
+                                    "TotalOtherTaxesAmt23" => nil,
+                                    "TotalPaymentsAmt33" => "0",
+                                    "TotalTaxAmt24" => nil,
+                                    "TotalTaxBeforeCrAndOthTaxesAmt18" => nil,
+                                    "TotalTaxablePensionsAmt5b" => nil,
+                                    "WagesSalariesAndTipsAmt1" => nil,
+                                    "WithholdingTaxAmt25d" => nil,
+                                    "ZIPCd" => "",
+       })
+      end
+    end
+
+    context "with a filled out submission record" do
+      before do
+        submission.intake.update(primary_ip_pin: "12345", primary_signature_pin_at: Date.new(2020, 1, 1))
+        submission.reload
+      end
+
+      it "returns a filled out pdf" do
+        output_file = pdf.output_file
+        result = non_preparer_fields(output_file.path)
+        expect(result).to match(hash_including(
+          "FilingStatus" => "1",
+          "PrimaryFirstNm" => submission.intake.primary_first_name,
+          "PrimaryLastNm" => submission.intake.primary_last_name,
+          "PrimarySSN" => "XXXXX#{submission.intake.primary_ssn.last(4)}",
+          "AddressLine1Txt" => "23627 HAWKINS CREEK CT",
+          "CityNm" => "KATY",
+          "StateAbbreviationCd" => "TX",
+          "ZIPCd" => "77494",
+          "VirtualCurAcquiredDurTYInd" => "false",
+          "TaxableInterestAmt2b" => "1",
+          "TotalIncomeAmt9" => "1",
+          "AdjustedGrossIncomeAmt11" => "1",
+          "TotalItemizedOrStandardDedAmt12" => "12400",
+          "TaxableIncomeAmt15" => "0",
+          "RecoveryRebateCreditAmt30" => submission.tax_return.claimed_recovery_rebate_credit.to_s,
+          "RefundableCreditsAmt32" => submission.tax_return.claimed_recovery_rebate_credit.to_s,
+          "TotalPaymentsAmt33" => submission.tax_return.claimed_recovery_rebate_credit.to_s,
+          "OverpaidAmt34" => "0",
+          "RefundAmt35" => submission.tax_return.claimed_recovery_rebate_credit.to_s,
+          "PrimarySignature" => "#{submission.intake.primary_first_name} #{submission.intake.primary_last_name}",
+          "PrimarySignatureDate" => "01/01/20",
+          "PrimaryIPPIN" => "12345",
```

Comment:
If we're masking the SSN, should we also mask the IP PIN? Maybe mask it down to 2 digits?

What's the purpose of the masking? When a client wants their full 1040 from an e-file preparer, I would assume it would contain the full information they provided so they can e.g. check for SSN errors.

The IP PIN is probably about as sensitive as the SSN, hence the question.

Reply:
Rae indicated that the 1040s they get from TaxSlayer have important information masked and that is the "safe" way to generate these. Re: other sensitive fields - product will review and make additional requests if necessary once i can get this in front of them, but i started with the fields Rae explicitly requested to be masked.